frontend/javascripts/oxalis/merger_mode.js

```diff
@@ -80,33 +85,32 @@ function getAllNodesWithTreeId(): Array<NodeWithTreeId> {
   return nodes;
 }
 
-/* Here we intercept calls to the "addNode" method. This allows us to look up the segment id at the specified
-   point and display it in the same color as the rest of the aggregate. */
-async function createNodeOverwrite(store, call, action, mergerModeState: MergerModeState) {
-  call(action);
+/* React to added nodes. Look up the segment id at the node position and
+  display it in the same color as the rest of the aggregate. */
+async function createNode(
+  mergerModeState: MergerModeState,
+  nodeId: number,
+  treeId: number,
+  position: Vector3,
+) {
   const { colorMapping, segmentationLayerName, nodeSegmentMap } = mergerModeState;
-  const pos = action.position;
-  const segmentId = await api.data.getDataValue(segmentationLayerName, pos);
+  const segmentId = await api.data.getDataValue(segmentationLayerName, position);
 
-  const activeTreeId = api.tracing.getActiveTreeId();
-  const activeNodeId = api.tracing.getActiveNodeId();
-  // If the node wasn't created. This should never happen.
-  if (activeTreeId == null || activeNodeId == null) {
-    Modal.info({ title: "The created node could not be detected." });
-    return;
-  }
   // If there is no segment id, the node was set too close to a border between segments.
   if (!segmentId) {
-    Modal.info({ title: "You've set a point too close to grey. The node will be removed now." });
-    api.tracing.deleteNode(activeNodeId, activeTreeId);
+    api.utils.showToast(
+      "warning",
+      "You've set a node too close to grey. The node will be removed now.",
```

Comment:
I am not a big fan of the warning text. What is the `grey`? I don't quite understand the message. 

(Also this message should probably belong into `messages.js`)

Reply:
I like your suggestion. Makes it a lot clearer and does not require intimate knowledge about internals and colors. 

```suggestion
      "You have placed a node outside of a segment. The node will be removed now.",
```